BraveWallet/Settings/CustomNetworkDetailsView.swift

```diff
@@ -0,0 +1,458 @@
+        for (index, item) in rpcUrls.enumerated() {
+          if item.input.isEmpty && item.error == nil { // no validation on new entry
+            hasNewEntry = true
+          } else {
+            if URIFixup.getURL(item.input) == nil {
+              rpcUrls[index].error = Strings.Wallet.customNetworkInvalidAddressErrMsg
+            } else {
+              rpcUrls[index].error = nil
+            }
+          }
+        }
+        // Only insert a new entry when all existed entries pass validation
+        if rpcUrls.compactMap({ $0.error }).isEmpty && !hasNewEntry {
+          rpcUrls.append(NetworkInputItem(input: ""))
+        }
+      }
+    }
+  }
+  @Published var iconUrls = [NetworkInputItem(input: "")] {
+    didSet {
+      // we only care the set on each item's `input`
+      if iconUrls.reduce("", { $0 + $1.input }) != oldValue.reduce("", { $0 + $1.input }) {
+        // validate every entry except the last new entry if there is one
+        var hasNewEntry = false
+        for (index, item) in iconUrls.enumerated() {
+          if item.input.isEmpty && item.error == nil { // no validation on new entry
+            hasNewEntry = true
+          } else {
+            if URIFixup.getURL(item.input) == nil {
+              iconUrls[index].error = Strings.Wallet.customNetworkInvalidAddressErrMsg
+            } else {
+              iconUrls[index].error = nil
+            }
+          }
+        }
+        // Only insert a new entry when all existed entries pass validation and there is no new entry
+        if iconUrls.compactMap({ $0.error }).isEmpty && !hasNewEntry {
+          iconUrls.append(NetworkInputItem(input: ""))
+        }
+      }
+    }
+  }
+  @Published var blockUrls = [NetworkInputItem(input: "")] {
+    didSet {
+      // we only care the set on each item's `input`
+      if blockUrls.reduce("", { $0 + $1.input }) != oldValue.reduce("", { $0 + $1.input }) {
+        // validate every entry except the last new entry if there is one
+        var hasNewEntry = false
+        for (index, item) in blockUrls.enumerated() {
+          if item.input.isEmpty && item.error == nil { // no validation on new entry
+            hasNewEntry = true
+          } else {
+            if URIFixup.getURL(item.input) == nil {
+              blockUrls[index].error = Strings.Wallet.customNetworkInvalidAddressErrMsg
+            } else {
+              blockUrls[index].error = nil
+            }
+          }
+        }
+        // Only insert a new entry when all existed entries pass validation and there is no new entry
+        if blockUrls.compactMap({ $0.error }).isEmpty && !hasNewEntry {
+          blockUrls.append(NetworkInputItem(input: ""))
+        }
+      }
+    }
+  }
+  
+  /// Updates the details of this class based on a custom network
+  func populateDetails(from network: BraveWallet.EthereumChain) {
+    let chainIdInDecimal: String
+    if let intValue = Int(network.chainId.removingHexPrefix, radix: 16) {
```

Comment:
You may want to check if `network.chainId` has a `0x` prefix or is a hex string before doing this since its possible to have hex numbers that use only decimals

Reply:
`removingHexPrefix` has the logic to check if it has a `0x` it removes `0x` only if the string starts with a `0x`

Reply:
@nuo-xu 

I'm confused about the overall logic then:

```swift
if let intValue = Int(network.chainId.removingHexPrefix, radix: 16) {
  chainIdInDecimal = "\(intValue)"
} else {
  chainIdInDecimal = network.chainId
}
```

Is the value in `BraveWallet.EthereumChain` always hex? If it isn't then if you get a value like `10` are you assuming its a decimal value 10 instead of `0x10`? In the current logic it will succeed at parsing via `Int(_:radix:)` and result in `16` and not `10`

Reply:
yes we assume chainId is always in hex with or without a `0x`. We will always try to convert it to an integer. 
I double checked android code they are doing the same. 
<img width="545" alt="Screen Shot 2022-02-11 at 5 00 34 PM" src="https://user-images.githubusercontent.com/1187676/153676475-665f2072-4a05-4a36-a6a9-ac84264eee3d.png">

Reply:
It seems like Android has slightly different flow from that screen cap: They don't set the value of the text field if `chainId` can't be parsed as a hex value. Should we follow that same flow and do nothing to `chainIdInDecimal` if it `Int(_:radix:)` returns nil?

Reply:
Either way this seems fine, maybe toss a comment for future me/you about EthereumChain always being hex

Reply:
I'd like display whatever we get from `BraveWallet.EthereumChain.chainId` if we can parse it to hex. It would be weird that the users try to edit the network but the network has no id? with no id users won't be able to update anyways. 
I have added a comment saying  `BraveWallet.EthereumChain.chainId` always in hex.